src/to/toCsv.ts

```diff
@@ -0,0 +1,43 @@
+import { Analysis } from '../Analysis';
+import type { SelectorType, SpectrumType } from '../types';
+
+export function toCsv(analysis: Analysis, options?: SelectorType) {
+  // Export all the data to Csv
+  if (!options) {
+    return exportCSV(analysis.spectra);
+  }
+
+  // Export selected variables
+  const spectrums = analysis.getXYSpectrum(options);
+  return exportCSV(spectrums ? [spectrums] : []);
+}
+
+function exportCSV(spectrums: SpectrumType[]) {
+  let res: string[] = new Array(spectrums.length);
+  for (let index = 0; index < spectrums.length; index++) {
+    const variables = Object.values(spectrums[index].variables);
+
+    const columns = variables.reduce(
+      (acc: { max: number; variables: string[] }, curr) => {
+        const max = Math.max(acc.max, curr.data.length);
+        const variables = [...acc.variables, curr.label];
+        return { max, variables };
+      },
+      { max: 0, variables: [] },
```

Comment:
I did code simplification. We should not use 'reduce' instead of a loop.
(and in this case a map is even simpler and easier to unsertand)